hyperspy/utils/peakfinders2D.py

```diff
@@ -27,6 +27,37 @@
 
 NO_PEAKS = np.array([[np.nan, np.nan]])
 
+def _get_peak_position_and_intensity(X, f, **kwargs):
+    """
+    Take some function, f, and apply it to the 2d array X returning a list
+    of peak positions.  The intensity at each peak position is then appended
+    to the returned list of peaks
+
+    Parameters
+    ----------
+    X: 2-D array-like
+        The input image used to find peaks
+    f: func
+        The passed function to find peaks
+    kwargs:
+        Any additional keyword arguments passed to f
+
+    Returns
+    -------
+        peaks:array-like
+            A 2d array with columns [x, y, intensity]
+
+    """
+    peaks = f(X, **kwargs)
+
+    if np.any(np.isnan(peaks)): #handle no peaks
+        return np.array([[np.nan, np.nan,np.nan]])
+    else:
+        rounded_peaks = np.round(peaks).astype(int)
+        maxima = tuple([tuple(rounded_peaks[:,d]) for d in range(2)])
+        intensity = X[maxima]
```

Comment:
I find easier the following easier to follow/read rather using list comprehension and creating tuple of tuples in combination with numpy advanced indexing:
```suggestion
        peaks_indices = np.round(peaks).astype(int)
        intensity = X[peaks_indices[:, 0], peaks_indices[:, 1]]

```